database_sanitizer/dump/postgres.py

```diff
@@ -90,35 +80,41 @@ def sanitize(url, config):
         current_table = copy_line_match.group("table")
         current_table_columns = parse_column_names(copy_line_match.group("columns"))
 
-        sanitize_values = get_sanitizer(
+        sanitize_value_line = get_value_line_sanitizer(
             config, current_table, current_table_columns)
 
         yield line
 
 
-def get_sanitizer(config, table, columns):
+def get_value_line_sanitizer(config, table, columns):
     if not config:
         return None
 
-    sanitizers = [
-        config.get_sanitizer_for(table, field)
-        for field in columns
-    ]
+    def get_sanitizer(column):
+        sanitizer = config.get_sanitizer_for(table, column)
 
-    if not any(sanitizers):
-        return None
+        if not sanitizer:
+            return _identity
 
-    sanitizers_or_identities = [
-        (sanitizer or _identity) for sanitizer in sanitizers
-    ]
+        def decode_sanitize_encode(value):
+            return encode_copy_value(sanitizer(decode_copy_value(value)))
+
+        return decode_sanitize_encode
+
+    sanitizers = [get_sanitizer(column) for column in columns]
+
+    if all(x is _identity for x in sanitizers):
+        return None
```

Comment:
This early return seems to be untested now. Should probably add a test case for this.

Reply:
Now it too is tested.